src/net.cpp

```diff
@@ -117,6 +117,32 @@ void CConnman::AddAddrFetch(const std::string& strDest)
 
 uint16_t GetListenPort()
 {
+    // If -bind= is provided with ":port" part, use that (first one if multiple are provided).
+    for (const std::string& bind_arg : gArgs.GetArgs("-bind")) {
+        constexpr uint16_t dummy_port = 0;
+        constexpr bool allow_lookup = false;
+        CService bind_addr;
+
+        if (Lookup(bind_arg, bind_addr, dummy_port, allow_lookup)) {
+            if (bind_addr.GetPort() != dummy_port) {
+                return bind_addr.GetPort();
+            }
+        }
+    }
+
+    // Otherwise, if -whitebind= without PF_NOBAN is provided, use that
+    // (-whitebind= is required to have ":port").
+    for (const std::string& whitebind_arg : gArgs.GetArgs("-whitebind")) {
+        NetWhitebindPermissions whitebind;
+        bilingual_str error;
+        if (NetWhitebindPermissions::TryParse(whitebind_arg, whitebind, error)) {
+            if (!(whitebind.m_flags & PF_NOBAN)) {
```

Comment:
Good catch!

Previously, if `-whitebind=download@1.1.1.1:8765` was specified we would not have returned `8765` from `GetListenPort()` but we should have.

Reply:
Yes, I've been working on this issue with #21644 that fixes the same issue and #21506 that would eliminate this form of `NetPermissionFlags` bug.

Can you please use the `NetPermissionFlags::` scope as suggested in https://github.com/bitcoin/bitcoin/pull/20196#discussion_r610770148? This is already the case in other places and it would avoid an unnecessary future commit (or rebase of #21506).